testsuite/ui/views/devel/settings/braintree.py

```diff
@@ -59,18 +62,35 @@ def _iframe_fill(self, frame_id, widget, value):
         self.browser.selenium.switch_to.default_content()
 
 
-class OTPForm(View):
+class ChallengeForm(View):
     """3DS Braintree verification form"""
     FRAME = '//iframe[@id="Cardinal-CCA-IFrame"]'
     otp_input = TextInput(name="challengeDataEntry")
-    submit = ThreescaleSubmitButton()
+    submit_btn = ThreescaleSubmitButton()
+    cancel_btn = GenericLocatorWidget(".//input[@value='CANCEL']")
 
-    def complete_auth(self):
-        """Complete authentication"""
-        self.otp_input.wait_displayed()
+    def complete(self):
+        """Complete authentication challenge"""
+        self.otp_input.wait_displayed(timeout="20s")
         self.otp_input.fill("1234")
-        self.browser.element(self.submit).submit()
+        self.browser.element(self.submit_btn).submit()
         self.browser.selenium.switch_to.default_content()
+        self._wait_challenge_completed()
+
+    def cancel(self):
+        """Cancel authentication challenge"""
+        self.otp_input.wait_displayed(timeout="20s")
+        self.cancel_btn.click()
+        self.browser.selenium.switch_to.default_content()
+        self._wait_challenge_completed()
+
+    def _wait_challenge_completed(self):
+        """Wait for `BraintreeCCView` to reflect all changes when 3DS challenge was completed (or closed)"""
+        self.browser.wait_for_element(
+            ".//dt[normalize-space(.)='Credit card number']|"
+            ".//p[@class='alert alert-danger']",
+            timeout=20
```

Comment:
The dialog's default timeout is 60 seconds (see https://cardinaldocs.atlassian.net/wiki/spaces/CC/pages/1409568/Configurations#Configurations-RootLevelConfiguration). One of the testing credit cards is used in this scenario so I would at least pass `timeout` as an argument.

Reply:
Great point, I will add this scenario in the next PR (this one is already large)